src/chart/components/chart/chart.model.ts

```diff
@@ -1069,6 +1094,67 @@ export class ChartModel extends ChartBaseElement {
 			reindexCandles(series.dataPoints);
 			series.recalculateVisualPoints();
 		});
+		this.candlesChangedRedraw();
+	}
+
+	/**
+	 * Remove candles by ids and recalculate indexes
+	 * @param ids - candles ids to remove
+	 * @param selectedCandleSeries - candle series to remove candles from
+	 * @param isSequence - true, if candles follow one by one
+	 */
+	public removeCandlesByIds(ids: Candle['id'][], isSequence: boolean, selectedCandleSeries: CandleSeriesModel) {
+		if (!isSequence) {
+			ids.forEach(id => this.removeCandleByIdx(this.candleIdxFromId(id, selectedCandleSeries)));
+			return;
+		}
+
+		const firstIdx = this.candleIdxFromId(ids[0], selectedCandleSeries);
+		const lastIdx = this.candleIdxFromId(ids[ids.length - 1], selectedCandleSeries);
+
+		if (firstIdx < 0 || lastIdx < 0) {
+			console.warn('Candle is not found');
+			return;
+		}
+
+		// if array is a sequence the removal process should be faster because reindex and recalculate visual points would call only once
+		selectedCandleSeries.dataPoints = selectedCandleSeries.dataPoints
+			.slice(0, firstIdx)
+			.concat(selectedCandleSeries.dataPoints.slice(lastIdx + 1));
+
+		reindexCandles(selectedCandleSeries.dataPoints, lastIdx);
+		selectedCandleSeries.recalculateVisualPoints();
+
+		this.candlesChangedRedraw();
+	}
+
+	/**
+	 * Add candles by id and recalculate indexes
+	 *
+	 * @param candles - candles to add
+	 * @param startId - target candle to start adding candles from
+	 * @param selectedCandleSeries - candle series to add candles to
+	 */
+	public addCandlesById(candles: Candle[], startId: string, selectedCandleSeries: CandleSeriesModel) {
```

Comment:
btw, why do we need all those methods now?
How we lived without them and it was pretty fine?